### ISSUE TITLE
What's the health status of the project/some of the plugins?

### ISSUE THREAD

**Issue by svenstaro** [unbit/uwsgi#2287]
Hello,

I don't mean to provoke with the title but looking at the recent commits, open issues, pull requests as well as a look at the [last commit dates for many of the plugins](https://github.com/unbit/uwsgi/tree/master/plugins) I'm left wondering whether the project is currently in good shape.

For context, I'm the maintainer of the official Arch Linux uwsgi package and we have an interest in keeping uwsgi working and in repos but I'm especially skeptical about the maintenance state of some of the plugins in uwsgi. In particular, I'm curious whether we can expect compatibility fixes going forward (for instance, right now PHP8 is incompatible).

It seems like uwsgi's Python plugin is best maintained while all of the other plugins don't appear to receive nearly as much love. I don't want to package potentially unmaintained software (considering the security implications) and would like to hear some kind of recommendation from the current maintainers of uwsgi whether I should bother with packaging most of the plugins or whether we're best off only using the Python plugin(s).

**Comment by xrmx** [comment]
Fair question :) While Python plugin is widely used I really don't know how much popular other language plugins are. As a distribution packager you probably have more clue on how a package is popular, any chance you have some stats? In Debian it looks like they are not used much https://qa.debian.org/popcon.php?package=uwsgi

**Comment by svenstaro** [comment]
In Arch we have this: https://pkgstats.archlinux.de/packages#query=uwsgi

So it would seem that the Python plugin is, as suspected, by far the most popular, followed with some distance by PHP.

Anyhow, my question was not really about the popularity of these plugins but more about how much we can expect development efforts for those in the future. For instance, now with PHP8 not compiling with uwsgi-plugin-php, are we expected to fix that ourselves (probably no resources for that) or can we expect upstream to fix it or should we consider dropping it?

**Comment by xrmx** [comment]
Given the development resources are scarce better spend time on fixing used stuff than unused ones :) It looks like php plugin is worth fixing then.
To anyone reading this if you care about uwsgi feel free to reach out to sponsors my time for fixing stuff. 

ps. php 8 build should be fixed there https://github.com/unbit/uwsgi/pull/2288

**Comment by dvzrv** [comment]
@xrmx as a user of the php plugin for uwsgi: many thanks!

**Comment by dvzrv** [comment]
> To anyone reading this if you care about uwsgi feel free to reach out to sponsors my time for fixing stuff.

Do you use e.g. github sponsors? It would probably be a good idea to promote this a bit!
Not sure whether a lot of people are willing to sponsor, but promoting this topic at least would make it immanently visible.

**Comment by xrmx** [comment]
@dvzrv Tried yesteday to setup but stripe (requirement for payments) didn't like any of my documents :shrug: , will try again

**Comment by svenstaro** [comment]
Any chance for a new release? Then I'd consider this issue closed. Also perhaps you can also remove some super old and unmaintained plugins.

**Comment by xrmx** [comment]
@svenstaro I suggest to cherry-pick the patches and bump the arch local version

**Comment by svenstaro** [comment]
Of course, and we've done so. But on top of that, a new uwsgi would still be nice since the last release is fairly long ago in software terms. :)

**Comment by dvzrv** [comment]
In hindsight of https://github.com/unbit/uwsgi/issues/2394 (uwsgi is incompatible with php 8.1), we need to reiterate on this topic again.
It has become clear, that this issue does not solve itself and that the resources that can be spent on uwsgi are not sufficient for it to be usable with up-to-date technology unfortunately.

As this is a continued problem I am currently contemplating to remove uwsgi from the official repositories on Arch Linux (although I make use of it myself for various PHP applications), as the negative impact on other packages relying on it is too high to justify it being in the repositories and used (for PHP at least). :cry:

**Comment by jSML4ThWwBID69YC** [comment]
@xrmx 

Any luck with the Stripe issue so people can donate?

**Comment by dvzrv** [comment]
@xrmx any news on this topic?

FWIW, the following funds might be of interest for you to apply to: 
* https://nlnet.nl/
* https://sovereigntechfund.de/en.html
